### PR TITLE
update new box handling - remove fetch all boxes dependency

### PR DESCRIPTION
**Pull request by JoeArmani**
Related: Phase 2 Biospecimen Dashboard updates - https://github.com/episphere/connect/issues/708

We've updated box handling to use an incrementing data point in the 'siteDetails' collection for new box creation and numbering. Prior to this, the app has been dependent on pulling all boxes and searching for the highest box number. It's important to get away from needing all box data for overall app data handling efficiency. This is one step towards not needing to fetch that data.

### CODE REVIEW

File: src/pages/shipping.js [hunk 1]
```diff
@@ -314,87 +314,119 @@ const handleRemoveBagButton = (currDeleteButton, currTubes, currBoxId) => {
...
+        boxToAdd[conceptIds.shippingBoxId] = 'Box' + largestBoxNum.toString();
+    
+        try {
+            const addBoxResponse = await addBoxAndUpdateSiteDetails(boxToAdd);
+            if (addBoxResponse.code === 200) {
+                boxList.push(boxToAdd);
+                return boxToAdd;
+            } else if (addBoxResponse.code === 409) {
+                attempts ++;
+                await delayRetryAttempt(1000);
```
Comment (Gbarra9) [inline src/pages/shipping.js:394]: I like the one second delay for checking whether the next box# increment exists or not

**Review by Gbarra9 — APPROVED** [review]
Looks good. Added a few small comments about spacing. 

I like the implementation of adding a new box. More specifically, in the while loop you added logic to continue incrementing if the incremented `Box#` already exists. You also gave a 1 second delay as well for possible increment attempts.

**Comment by JoeArmani** [comment]
Thanks, @Gbarra9, I appreciate the review and committed your suggested changes.

File: src/pages/shipping.js (hunk 1 above)
Comment (JoeArmani) [reply src/pages/shipping.js:394]: Thanks!

**Review by anthonypetersen — APPROVED** [review]
awesome!

**Pull request merged**